cpp/cmake/FindARROW.cmake

```diff
@@ -18,6 +18,7 @@ find_library(ARROW_LIB arrow REQUIRED)
 message(STATUS "Found arrow library at ${ARROW_LIB}")
 
 find_path(ARROW_INCLUDE_DIR arrow/type.h)
+include_directories("${ARROW_INCLUDE_DIR}")
```

Comment:
we are following the recommendations for modern/modular cmake, which says that all include_directories must be against a target only (within the project), and prefers taking dependency on targets (for external projects) instead of explicitly adding include-dirs/compile-flags/libs/..

https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1#forget-the-commands-add_compiler_options-include_directories-link_directories-link_libraries  

What's the reason for this change ? Maybe, I could suggest alternatives.

Reply:
Interesting. I added this because cmake was unable to find arrow includes without it.

Reply:
is this a problem while building gandiva ? Or, a project/test that depends on gandiva ? If it's the latter,  you could use the procedure we do for our integration tests.

  target_link_libraries(${TEST_NAME} PRIVATE gandiva)

Reply:
This is a problem when building gandiva. I have arrow installed in a conda environment. I can provide additional steps to reproduce if needed.

Reply:
sorry for the delay, I was on vacation. Can you please provide the steps to reproduce ?

Also, can you please spin off a different PR for the memleak fix ?

Reply:
Yep. I should be able to get to this tonight.